app/assets/javascripts/components/timeline/timeline.jsx

```diff
@@ -280,18 +296,37 @@ const Timeline = createReactClass({
       wizardLink = <CourseLink to={wizardUrl} className="button dark button--block timeline__add-assignment">Add Assignment</CourseLink>;
     }
 
-    const controls = this.props.reorderable || __guard__(this.props, x => x.editableBlockIds.length) > 1 ? (
+    const saveChangesButton = (
+      <button className="button dark button--block" onClick={this.props.saveGlobalChanges}>
+        {I18n.t('timeline.save_all_changes')}
+      </button>
+    );
+    const cancelChangesButton = (
+      <button className="button button--clear button--block" onClick={this.props.cancelGlobalChanges}>
+        {I18n.t('timeline.discard_all_changes')}
+      </button>
+    );
+    const resetTitlesButton = (
+      <button className="button button--clear button--block" onClick={this.props.resetTitles}>
+        {I18n.t('timeline.reset_titles')}
+      </button>
+    );
+    const reorderActionButtons = this.props.reorderable || __guard__(this.props, x => x.editableBlockIds.length) > 1 ? (
       <div>
-        <button className="button dark button--block" onClick={this.props.saveGlobalChanges}>
-          {I18n.t('timeline.save_all_changes')}
-        </button>
-        <button className="button button--clear button--block" onClick={this.props.cancelGlobalChanges}>
-          {I18n.t('timeline.discard_all_changes')}
-        </button>
+        {saveChangesButton}
+        {cancelChangesButton}
+      </div>
+    ) : undefined;
```

Comment:
This is minor, but for an intentional "nothing" value, I would typically use `null`.